osdep/threads-posix.h

```diff
@@ -130,29 +134,68 @@ static inline int mp_mutex_init_type_internal(mp_mutex *mutex, enum mp_mutex_typ
 #define mp_mutex_trylock    pthread_mutex_trylock
 #define mp_mutex_unlock     pthread_mutex_unlock
 
-#define mp_cond_init(cond)  pthread_cond_init(cond, NULL)
-#define mp_cond_destroy     pthread_cond_destroy
-#define mp_cond_broadcast   pthread_cond_broadcast
-#define mp_cond_signal      pthread_cond_signal
-#define mp_cond_wait        pthread_cond_wait
+static inline int mp_cond_init(mp_cond *cond)
+{
+    if (!cond)
+        return EINVAL;
```

Comment:
Why `EINVAL` instead of `assert`? Or does mpv codebase actually call this with NULL intentionally?

Reply:
Same remarks on all the other `EINVAL` below. 99% of the time, `EINVAL` suggest a buggy code and should be assert-ed instead of returning an error code to a caller that's proven himself to be flaky.